### PR TITLE
wall of bricks using js function

### PR DESCRIPTION
**Pull request by cwagner**

### CODE REVIEW

File: bricks.js [hunk 1]
```diff
@@ -9,3 +9,56 @@ function borderclick(id) {
 	}
 
 } 
+
+function drawpuzzle(width, height) {
+	if (height % 2 == 0) {
+		console.log("Height of brick wall must be odd");
```
Comment (johnbcoughlin) [inline bricks.js:15]: use console.error()!

File: bricks.js [hunk 2]
```diff
@@ -9,3 +9,56 @@ function borderclick(id) {
...
+		return;
+	}
+	var svgns = "http://www.w3.org/2000/svg";
```
Comment (johnbcoughlin) [inline bricks.js:18]: i think you can put this directly in the svg tag in the HTML? I could be wrong though.

File: bricks.js [hunk 3]
```diff
@@ -9,3 +9,56 @@ function borderclick(id) {
...
+	for (var y = 0; y < height; y++) {
```
Comment (johnbcoughlin) [inline bricks.js:19]: `height` sounds like a length to me; I would call this `nRows` or something similar.

File: bricks.js [hunk 4]
```diff
@@ -9,3 +9,56 @@ function borderclick(id) {
 	}
 
 } 
+
+function drawpuzzle(width, height) {
```
Comment (johnbcoughlin) [inline bricks.js:13]: `nColumns`, `nRows`

File: bricks.js [hunk 5]
```diff
@@ -9,3 +9,56 @@ function borderclick(id) {
...
+	if (height % 2 == 0) {
+		console.log("Height of brick wall must be odd");
+		return;
+	}
+	var svgns = "http://www.w3.org/2000/svg";
+	for (var y = 0; y < height; y++) {
+		for (var x = 0; x < (width - y % 2); x++) {
+			var rect = document.createElementNS(svgns, 'rect');
+			var num = x + 1;
+			rect.setAttributeNS(null, 'x', 10 + 110 * x + 55 * (y % 2));
```
Comment (johnbcoughlin) [inline bricks.js:23]: pull out these numbers into constants. like `brickWidth = 110;`, etc.

File: bricks.js [hunk 6]
```diff
@@ -9,3 +9,56 @@ function borderclick(id) {
...
+	}
+	for (var y = 0; y < height + 1; y++) {
+		for (var x = 0; x < 2 * width; x++) {
+			var rect = document.createElementNS(svgns, 'rect');
+			var num = x + 1;
+			rect.setAttributeNS(null, 'x', 10 + 55 * x);
+			rect.setAttributeNS(null, 'y', 60 * y);
+			rect.setAttributeNS(null, 'height', 10);
+			rect.setAttributeNS(null, 'width', 45);
+			rect.setAttributeNS(null, 'id', 'L' + String.fromCharCode(65 + y) + num.toString());
```
Comment (johnbcoughlin) [inline bricks.js:41]: what is the L here? just curious.

File: bricks.js [hunk 7]
```diff
@@ -9,3 +9,56 @@ function borderclick(id) {
...
+	}
+	for (var y = 0; y < height; y++) {
+		for (var x = 0; x < (width + 1 - y % 2); x++) {
+			var rect = document.createElementNS(svgns, 'rect');
+			var num = x + 1;
+			rect.setAttributeNS(null, 'x', 110 * x + 55 * (y % 2));
+			rect.setAttributeNS(null, 'y', 10 + 60 * y);
+			rect.setAttributeNS(null, 'height', 50);
+			rect.setAttributeNS(null, 'width', 10);
+			rect.setAttributeNS(null, 'id', 'T' + String.fromCharCode(65 + y) + num.toString());
```
Comment (johnbcoughlin) [inline bricks.js:55]: and the T here :)

File: bricks.js [hunk 8]
```diff
@@ -9,3 +9,56 @@ function borderclick(id) {
...
+			rect.setAttributeNS(null, 'y', 60 * y);
+			rect.setAttributeNS(null, 'height', 10);
+			rect.setAttributeNS(null, 'width', 45);
+			rect.setAttributeNS(null, 'id', 'L' + String.fromCharCode(65 + y) + num.toString());
+			rect.setAttributeNS(null, 'class', 'unclicked');
+			rect.setAttributeNS(null, 'onclick', 'borderclick(this)');
+			document.getElementById("canvas").appendChild(rect);
+		}
+	}
+	for (var y = 0; y < height; y++) {
```
Comment (johnbcoughlin) [inline bricks.js:47]: I encourage you to add some comments before these nested loops, explaining what element of the grid you are about to add.

File: bricks.js [hunk 9]
```diff
@@ -9,3 +9,56 @@ function borderclick(id) {
...
+function drawpuzzle(width, height) {
+	if (height % 2 == 0) {
+		console.log("Height of brick wall must be odd");
+		return;
+	}
+	var svgns = "http://www.w3.org/2000/svg";
+	for (var y = 0; y < height; y++) {
+		for (var x = 0; x < (width - y % 2); x++) {
+			var rect = document.createElementNS(svgns, 'rect');
+			var num = x + 1;
```
Comment (johnbcoughlin) [inline bricks.js:22]: find a better name than `num`

File: bricks.js (hunk 8 above)
Comment (johnbcoughlin) [reply bricks.js:47]: even better, you could split up this function into three smaller functions, each informatively named. :)

File: bricks.html [hunk 10]
```diff
@@ -3,14 +3,8 @@
 	<link rel="stylesheet" href="bricks.css">
 </head>
 <body>
+<svg width="5000" height="3000" id="canvas">
```
Comment (johnbcoughlin) [inline bricks.html:6]: use `100%` for the width and the height instead.

**Review by johnbcoughlin — APPROVED** [review]

File: bricks.js (hunk 6 above)
Comment (cwagner) [reply bricks.js:41]: Ha it's for 'long' and T is for 'tall'? I couldn't think of anything better

File: bricks.js (hunk 2 above)
Comment (cwagner) [reply bricks.js:18]: Could you explain what you mean by this?

**Pull request merged**